### DISCUSSION TITLE
Add speed attribute to cover integration

### DISCUSSION THREAD

**Discussion by Bascht74** [home-assistant/architecture#789] [Entity Models] [answered]
### Context
Currently the cover integration supports many different types of covers and shutters. To control them, there are the specialized services for this, including `cover.set_cover_position` and `cover.open_cover`. These can be used to control and position the covers. 

In the meantime, more and more manufacturers are starting to offer two speeds for their cover motors. Mostly it is called "silent mode". An example is Somfy (RS 100 io) or also Elero (RolMotion). The sense behind it is that basically all automatic controls are done in "silent mode" (speed doesn't matter, noise does), while manual controls by humans can be done fast (speed does matter). 

Two examples: 
1. in the evening the covers should close in the whole house and all motors are controlled at the same time. This makes a lot of noise in the house. In Silent mode, this is hardly noticeable - despite all motors being controlled. Same in the morning, which prevents you from waking up due to the noise generated by all covers. 
2. a person wants to open the cover to the terrace. Of course, this should be done quickly and is therefore carried out at the fast speed.

Especially in apartment buildings with multiple households, the possibility of activating silent mode in the morning and evening is very useful, as this means that all people living there are not disturbed by the noise of the covers of their neighbors. In Germany, for example, DIN 4109 was created especially for this purpose (also includes sound insulation for electrically operated shutters). 

Current state in Home Assistant: 
The current cover integration allows in principle the control of the devices, but not the choice of the desired speed.

### Proposal
Extension of the cover integration by the possibility to select the speed (slow/fast) for all services (e.g. `open`, `close`, `set_cover_position`). 
(So far I have only seen these two speeds in practice, more levels don't really make sense either. The choice to make here is: silent oder fast, not to balance those two out like with fans)

Example for a service call:

```
    - service: cover.set_cover_position
      target:
        entity_id: cover.demo
      data:
        position: 50
        speed: slow/fast
```

If the "speed" attribute is not specified, the default speed is used. If speed is specified and the device integration does not support different speeds, the attribute is ignored.

If a `SUPPORT_SLOW_SPEED` flag is added as well it should do nothing to existing integrations that doesn't support different speeds.

### Consequences
Until now, integrations have to find their own way for speed selection. E.g. the official overkiz integration did exactly that: 
https://github.com/home-assistant/core/blob/58883feaf69f5204c73812283fc7509ce5cb3c0c/homeassistant/components/overkiz/cover_entities/vertical_cover.py#L168).

By including the speed in the cover integration, a uniform standard for it would exist and the standard HA services would be usable. 
Furthermore, including a speed attribute in the cover integration itself would create the possibility that other projects can include this either. 

So far, I have often observed that HA has jumped on topics concerning important issues such as energy. 
This proposal is, in effect, about noise abatement. Therefore, I hope that the proposal is really considered, as noise control is an important factor in people's quality of life :-)

P.S.: 
There is a older entry about it (https://github.com/home-assistant/architecture/discussions/602), but it was never really discussed to the end. As more and more producers are jumping in that train, and integrations start to implement it outside the cover integration, I think it makes sense to bring it up again and explain the purpose and advantages it a bit further.

**Comment by elupus** [comment]
I think this make sense. Can you summarize a list of integrations/motors that could make use of this type of addition?

**Reply by ntindle** [reply]
Looks like it's numerous integrations now having issues with this

**Comment by Bascht74** [comment]
Hi elupus, 

I think of the official integrations that a lot of people use, it would potentially be: 
- https://www.home-assistant.io/integrations/overkiz -> There are already covers that support two speeds (quiet/fast), and many people use this integratoin (myself included). E.g. the somfy RS 100 io motor supports two speeds. Maybe also motors from other manufacturers supported by overkiz, but I don't know them personally.
- https://www.home-assistant.io/integrations/homematic/ -> There is an option to set the speed of the motors for "automatic drives" and "manual drives", e.g. here https://www.contronics.de/download/dokumente/erfal-Homematic-IP-Akku-Funk-Antrieb-DE_EN.pdf.

for maker solutions it would be of course also possible, if the suitable hardware is available: 
- https://www.home-assistant.io/integrations/cover.command_line/ 
- https://www.home-assistant.io/integrations/cover.template/ 

Furthermore there is a potential for newer motors which can be controlled by esphome / Tasmota. The prerequisite would be that Tasmota / ESPHome supports it (a chicken and egg problem). 
One can use e.g. https://www.elero.com/en/products/electrical-drives/rolmotion-m 
for this. I have currently implemented it with Tasmota itself and added a custom command for it via Berry as I am switching from Somfy to this other brand.

There could be more if you count integrations that are not used by many people or custom integrations.
E.g. Hella offers such a motor as well: https://www.hella.info/de/produkte/onyx-r-silent-motor#Mehr-Komfort and for this there is an integration: https://github.com/muhlba91/onyx-homeassistant-integration

**Reply by ntindle** [reply]
We can add switchbot curtains to this list now

**Reply by lightheaded** [reply]
Namely, [SwitchBot Curtain 3](https://support.switch-bot.com/hc/en-us/articles/16806028444439-Information-on-the-Quietdrift-Feature-in-SwitchBot-Curtain-3)

**Reply by ScottG489** [reply]
SwitchBot Blind Tilt would be another device. It has an option in the native app for a "silent" option, which it sounds like would apply here.

**Comment by tetienne** [comment]
Such new attribute will really ease the Overkiz integration. Many people ask for the silent mode before we implemented it. First using a virtual switch (not allowed) then by duplicating a cover to force this mode and by exposing a specific service.

In addition to `cover.set_cover_position`, I will suggest also to add this attribute to `cover.open` and `cover.close`.

Device option doesn't exist yet, but it can also be a solution for this silent mode.

**Comment by quenthal** [comment]
This would also help enormously Switchbot Curtain 3 integration. With Curtain 3 (and for some extent, Curtain 2) support slower, but quieter opening/closing of curtain. This is already implemented in the backbone, but they need to be exposed as desribed here for the cover -methods as attributes.

**Comment by lightheaded** [comment]
Thanks for initiating the discussion! What's needed for this to move forward?

**Comment by gjohansson-ST** [comment]
I'm only wondering here if we should limit to a specific set of speeds, why not similar to `preset` maybe have some default ones but the implementation can expand/set what they need?

**Reply by otsoni** [reply]
I was thinking the same thing. But should it be some kind of limited set of enums or something or should there be possibility for numeral value also?

**Reply by gjohansson-ST** [reply]
Don't see a reason to include numerical speeds so I think this proposal should limit itself to `list[str]`.
More or less a copy of `preset`. I'll update my comment above to outline the proposal more extensively.

**Reply by sandervandenberg** [reply]
If a device manufacturer decided to implement a completely variable speed setting, you'd need to change the code again. Would love to see a numerical value (0-255?) and in addition to that a limited set of enums which just translate to that numerical value.
Then in the implementation for a specific device one can specify if it supports a numerical value, or enum values (and which), or both.

**Comment by codyc1515** [comment]
The same is true of Zigbee motor covers. These includes devices such as [Aqara Roller Shade Driver](https://www.aqara.com/en/product/roller-shade-driver-e1/) and [Aqara Curtain Driver](https://www.aqara.com/en/product/curtain-driver-e1-track-version/). However, I would suggest that this functionality should be a little more open ended as Aqara devices, and likely others, support low / medium / high speeds.

Many climate entities also have something similar like "Boost" / "Eco" modes aka. "Fast" / "Slow" fan modes so it would be good to retain flexibility in that regard.

**Comment by snoack** [comment]
Is there any plan to allow configuring a default speed either per device or globally? After all this is not like a fan that sometimes you want to run on high and sometimes on low, but usually if people prefer to open their curtains (or whatever cover) slowly/quietly or rather fast they prefer to do so all the time.

**Reply by sandervandenberg** [reply]
Not all the time... I'd like to open my curtains very slowly, like a wake-up-light, before I get up in the morning. But I'd like it to close faster when the sun starts shining into the window to keep temperature down. Or to open it quickly when I want to open the window behind it.

**Comment by Dennis-Otto** [comment]
Since this discussion has gotten a little quiet, I wanted to ask about the current status. 
Are there any plans to add a speed configuration to the cover entity?
What is currently needed for development to continue? 
Is there already an approximate implementation date so that we can benefit from it?

Thanks in advance! :)

**Comment by sanderlv** [comment]
Is there any progress on this?

**Reply by lachezar-gizdov** [reply]
Wandering the same, it's been awhile since any update?

**Comment by EDelsman** [comment]
I would really like this discussion to come to a conclusion. It's been here almost 2 years. The switchbot PR has been held back because of this. I now have 9 cover entities that would benefit, from Somfy, Switchbot and Aqara. If I read correctly the only thing left to choose is between an int and an enum. I would opt for an approach similar to fan, which allows for both presets and speeds. Consistency is nice, and the implementations can borrow from the example.

**Reply by samtate** [reply]
This is getting a little silly now, 4 months since the last sensible comment and still no decision, holding back already programmed useful features from devices people have.

**Reply by Bascht74** [reply]
Hi @bdraco, 
as a switchbot user and core developer, could you help us out here how to get to a decision in this case?
This would help some integration developers to implement this feature in a right way an not working around it. 

I hope that you don't feel offended because I tagged to you directly. 
However, we really need an impulse to continue or to end the whole thing.
So, I hope it might be OK in this case. Thank you in advance for your help & wisdom :-)

**Reply by bdraco** [reply]
Hi Bascht74,

I don't actually use SwitchBot in production, and I've only been helping out with it because of my involvement with Home Assistant's Bluetooth stack so I'm not the person to advocate for this as I don't have any devices that support a speed.

One thing I would note that is this were to move forward, I'd recommend implementing the speed as 1-100 or a percentage so control can be normalized across devices and users don't have to figure out if one cover calls it `slow`, one calls it `quiet`, one calls it `gradual`, another calls it `fast` etc.

I also note that there are multiple requests to enumerate which integrations could use this feature but that isn't explained in the opening text or how each integration could use it which is likely not helping to move this forward as anyone who might be interested has to go digging through this to understand how it could be used.

**Comment by PhillyGilly** [comment]
I would like to be able to pass SPEED to my Velux Blinds. I can't use the HA cover control for Bedroom Blinds because it opens/closes the blinds at DEFAULT which is FAST (and noisy), so I have to run the native Velux controls in parallel to open the blinds in the morning in SILENT MODE. I Can see the the SPEED variable in the py code, but I need the HA cover control to be able to accept the input.

**Comment by Landcross** [comment]
So, I just bought some motorized curtain drivers and was surprised about it not being able to select the speed from Home Assistant. And then found this thread.

Why is it taking 2 years to come to a conclusion on this? C'mon everyone. We're simply talking about a cover speed here, not some plan to fire a rocket to the moon. Excuse my bluntness, but that's how I view it from someone not really familiar with HA Architecture discussions 😅

From reading the thread above, I quite like the idea from @bdraco on using a percentage and letting specific integrations handle the translation from percentage to whatever it is that device supports. Someone also mentioned the Fan entity. As far as I understand a percentage setting makes it also alligned with the speed percentage in the Fan entity. In fact, like @EDelsman mentioned, can't we just use a similar setup as [the Fan entity](https://developers.home-assistant.io/docs/core/entity/fan/)?  With a speed percentage and optional preset modes?

Again, I'm _definitly_ not an expert on this so take my thoughts with a tub of salt, but to me that sounds like the best way forward: configurable and also alligned with the setup of other entities. No need to reinvent the wheel right?

Who do we need to make a final decision on this?

**Reply by bdraco** [reply]
> Who do we need to make a final decision on this?

The proposal should be updated to incorporate the feedback from this discussion and include a summary of the integrations that could benefit from it. After that, a rough prototype implementation should be created to validate whether the proposed approach is viable.

We also need to explore if there is only a speed needed for open/close operations or if any integration allows controlling the tilt speed as well.

**Reply by bdraco** [reply]
@Bascht74 I think this is a good proposal and once we could likely approve once its updated to include the above.

**Reply by wollew** [reply]
So, to summarize, the proposal would be to add both speed presets and percentage to open/close/set_position and possibly tilt operations. As in the Fan entity, there would be optional `CoverEntityFeature.SET_SPEED` for percentages and `CoverEntityFeature.PRESET_MODE` for presets. 

Integrations mentioned so far that could support this feeature are:
* Aqara
* Overkiz / Somfy
* Homematic
* Switchbot
* Velux (which is why I came here, I recently opened https://github.com/home-assistant/core/pull/151594 without knowing about this proposal)
* a few custom integrations
* anything else?

**Comment by Gunni** [comment]
Throwing myself into the hat for switchbot curtain 3 speed control.

**Comment by wollew** [comment]
## Updated Proposal: Cover Speed Control

Hi everyone,

I'd like to help move this proposal forward. After reading through the discussion and [@bdraco](https://github.com/bdraco)'s feedback, here's an updated proposal incorporating the input so far.

## Summary

Add speed control to the Cover integration, similar to how the Fan entity handles speed percentage and preset modes. This allows users to control motor speed (silent/slow vs fast) for automated and manual operations.

## Integrations That Would Benefit

Integrations known so far:

- **Velux** - KLF 200 covers support velocity control (Default, Silent, Fast) and tilt speed control
- **SwitchBot** - Curtain 3 and Blind Tilt - supports silent mode in native app
- **Aqara Roller Shade Driver E1 & Curtain Driver E1** - Supports low/medium/high speeds
- **Somfy/Overkiz** - RS 100 io and other motors with silent mode
- **Homematic IP** - Separate speeds for automatic vs manual drives
- **Elero RolMotion** 
- **Hella Onyx R Silent Motor** - Via custom integration
- **Zigbee motor covers** - Various devices with multiple speed settings

## Proposed API Design

Following the Fan entity pattern with both percentage and preset modes:

### New Entity Features
```python
class CoverEntityFeature(IntFlag):
    # ... existing features ...
    SET_SPEED = 256  # Supports percentage-based speed (1-100)
    PRESET_MODE = 512  # Supports named preset modes
```

### New Entity Attributes
```python
class CoverEntity(Entity):
    _attr_speed: int | None = None  # Current speed percentage (1-100)
    _attr_preset_mode: str | None = None  # Current preset mode
    _attr_preset_modes: list[str] | None = None  # Available presets
    
    @property
    def speed(self) -> int | None:
        """Return the current speed percentage."""
        return self._attr_speed
    
    @property
    def preset_mode(self) -> str | None:
        """Return the current preset mode."""
        return self._attr_preset_mode
    
    @property
    def preset_modes(self) -> list[str] | None:
        """Return available preset modes."""
        return self._attr_preset_modes
```

### Common Preset Modes
While integrations can define their own presets, suggested common ones:
- `slow` / `silent` - For quiet operation
- `medium` / `normal` - For normal operation
- `fast` - For fast operation

### Service Updates

Add optional `speed` and `preset_mode` parameters to:
- `cover.open_cover`
- `cover.close_cover`
- `cover.set_cover_position`
- `cover.open_cover_tilt` (if tilt speed is supported)
- `cover.close_cover_tilt` (if tilt speed is supported)
- `cover.set_cover_tilt_position` (if tilt speed is supported)

Add new services:
- `cover.set_speed` - Set speed percentage (1-100)
- `cover.set_preset_mode` - Set preset mode by name

### Example Service Calls
```yaml
# Using percentage
service: cover.set_cover_position
target:
  entity_id: cover.bedroom_curtain
data:
  position: 50
  speed: 30  # 30% speed 

# Using preset mode
service: cover.close_cover
target:
  entity_id: cover.living_room_blind
data:
  preset_mode: silent

# Setting fast speed as preset
service: cover.set_preset_mode
target:
  entity_id: cover.bedroom_curtain
data:
  preset_mode: fast

# Use previously set speed/preset or default
service: cover.open_cover
target:
  entity_id: cover.bedroom_curtain

# Set speed without position change
service: cover.set_speed
data:
  speed: 50  # 50% speed 
target:
  entity_id: cover.bedroom_curtain
```

## Tilt Speed Support

Some integrations (Velux, Aqara, potentially others) may support tilt speed control as well. The proposal includes optional `speed` and `preset_mode` parameters for tilt operations.

## Backward Compatibility

- All new parameters are optional
- If `speed` or `preset_mode` are not specified, use device default or previously set values (speed/preset)
- Integrations without `SET_SPEED` or `PRESET_MODE` features ignore these parameters
- No breaking changes to existing integrations

## Benefits

- **Consistent API** across all cover integrations
- **Removes need for workarounds** (duplicate entities, separate select entities, custom services)
- **Aligns with existing patterns** (Fan entity)

## Next Steps

I'm willing to:
1. Create a prototype PR implementing this in the Cover base entity 
2. Update the Velux integration to use this pattern 
3. Work with other integration maintainers (Overkiz, etc.) to validate the approach

## Help needed

- While I can implement the base entity changes and update Velux in core, I would need help from someone who knows the frontend to update the UI to support speed and preset mode controls.

## Open questions for feedback:

1. Should percentage and preset modes be mutually exclusive, or can they coexist (like Fan)? If they coexist, are we suggesting the same mechanism as in the Fan entity to map presets to percentage values and vice versa?
2. Are there any edge cases we should consider for the prototype?
3. Is this the right place or does an updated proposal need a separate discussion thread? 

Looking forward to feedback!

**Comment by Orange-GT3** [comment]
Aqara Roller Shade Driver E1 user here, adding my support.

**Comment by PhillyGilly** [comment]
Great piece of work. KLF Velux user here. Strongly supported.

**Comment by emontnemery** [comment]
From the updated proposal
> Should percentage and preset modes be mutually exclusive, or can they coexist (like Fan)? If they coexist, are we suggesting the same mechanism as in the Fan entity to map presets to percentage values and vice versa?

What mechanism does the fan entity have which maps presets to percentage values? My understanding is that fan presets are intended for something more complex than fixed speeds, for example some automatic mode influenced by temperature.
Fixed speeds should instead be exposed via the numerical 0..100 control.

As an example, I don't think this is a correct way to use fan preset modes:
https://github.com/home-assistant/core/blob/e6b9c2f737fca426652c33e8cc1a52359a20599a/homeassistant/components/netatmo/fan.py#L49-L50

This is correct however:
https://github.com/home-assistant/core/blob/e6b9c2f737fca426652c33e8cc1a52359a20599a/homeassistant/components/rabbitair/fan.py#L34-L38

**Reply by wollew** [reply]
Indeed, I think I misread the Fan documentation, presets are not (or rather: should not be) used for speeds at all. This would mean covers could get rid of the presets completely and use the same mechanism as fans, using `speed_count` and `percentage`, mapping names to percentage via HA utility functions if necessary. If I understand it correctly, doing this would not expose the "names" (i.e. silent, fast, etc.) of the speeds at all, which are mostly prescribed what the cover devices support and might be what users are currently used to in the "native" app of their devices.

The alternative would be to use presets in a completely different way (using them for speed presets) than what fan does and maybe don't support percentage at all. I am not sure if any covers actually support a wide enough range of speeds to justify implementing percentage.

**Comment by wollew** [comment]
## Simplified Proposal: Device-Defined Speed Support

Following up on [this comment](https://github.com/home-assistant/core/pull/163819#issuecomment-4833896977) I updated the above proposal with a simplified version that only adds an optional speed parameter to cover movement services. I'd appreciate a short ack that this is what was agreed upon and I would then open a PR that extends the existing cover entity according to the suggestion below.

### Summary
Integrations define what speeds their devices support, and Home Assistant exposes those choices consistently through optional parameters on standard operations.

### Entity Changes
```python
class CoverEntity:
    @property
    def supported_speeds(self) -> list[str] | None:
        """Return list of speeds this device supports (e.g., ["silent", "fast"])
             for open/close/set_position operations.
        
        If None or [], device doesn't support speed selection.
        """
        return None
```

### Service Changes
Add optional `speed` parameter (string) to:
- `cover.open_cover`
- `cover.close_cover`  
- `cover.set_cover_position`

### State Attributes
`supported_speeds` must be included in the entity's state attributes so the frontend can read the available speeds

### Validation
- If `speed` is not in `supported_speeds`, raise `ServiceValidationError`
- If `speed` is None/not provided, use device default
- Unknown speeds are rejected by Home Assistant before reaching the integration

### Service Call Example
```yaml
service: cover.open_cover
target:
  entity_id: cover.bedroom_blind
data:
  speed: "silent"  
```

### Advantages
1. **No new services** - Uses only parameter additions
2. **Device-driven** - Each integration defines its own speed names naturally
3. **No normalization burden** - Home Assistant doesn't need to map between vendor terminology
4. **Simple validation** - Frontend/core can validate against `supported_speeds` list
5. **Backward compatible** - Existing code ignores the parameter entirely
6. **Extensible** - Future devices with more speeds need no architecture changes

**Reply by Gunni** [reply]
> Unknown speeds are rejected by Home Assistant before reaching the integration

Why not just try it and handle errors from integration?

**Reply by wollew** [reply]
> Why not just try it and handle errors from integration?

The idea is that integrations define their `supported_speeds`, so why would an integration not define a speed and then again validate? I mean, they still need to handle unknown things, but it should not be the normal case.

**Reply by iMicknl** [reply]
I am not sure if I understand your proposal.

> If empty list, device supports speed but only has default behavior.

Can you give an example here?

>["silent", "fast"]

Do integrations need to register the 'default' as well? Or is an empty value for speed the default? Not sure if you look at this like a preset or all options.

And with this proposal, how would this work for devices that support cover and tilt controls? Will this speed be applied to both?

**Reply by wollew** [reply]
> Can you give an example here?

The idea was, if `supported_speeds == []`, then you can pass an empty value as speed parameter, which will be equivalent to using the device's default. Which, thinking about it some more, probably doesn't make a lot of sense, so both None and [] should be treated as "does not support speed" probably.

> Do integrations need to register the 'default' as well?

Yes, I think this makes sense if only to be able to use the devices "native" naming. Although leaving the parameter of a movement service call empty will pick the default for the movement operation, whether it is registered or not. 
 
> And with this proposal, how would this work for devices that support cover and tilt controls?

Yes, the speed parameter can be passed to both open/close and open_tilt/close_tilt. Maybe we could have `supported_speeds` and `supported_tilt_speeds` in case there are devices that support different speeds for different operations.

**Reply by PhillyGilly** [reply]
I'm sorry that I have given up hope of ever seeing any proper progress in this since my initial involvement in November 2024 and have now resorted to making my own custom component that just makes my Velux blinds run in Silent mode instead of the default Fast (e.g. noisy!) The hardest part was to get the VELOCITY to enumerate, the rest was just changing the open and close functions. I could share files via my github if anybody is interested.

**Reply by iMicknl** [reply]
@wollew I think it would make sense indeed to have`CoverEntityFeature.SPEED` (or `SPEED_MODE`) and `CoverEntityFeature.TILT_SPEED`, so that it can be set for position vs tilt_position indiviually.

**Reply by wollew** [reply]
I've updated the proposal to have a separate `supported_tilt_speeds`, defaulted to return `supported_speeds`.

**Reply by gjohansson-ST** [reply]
This looks like the correct approach. There's no need for additional feature flags, a property can tell if it's supported and what options can be used. Validation should be taken care of in the `cover` component.

Two remarks
1. An empty list would be the same as `None` (no speed options available)
2. `supported_tilt_speeds` should not by default point to `supported_speeds`, integrations should handle what options there are (if any)

**Reply by wollew** [reply]
> Two remarks 

Ok, proposal updated.

**Reply by joostlek** [reply, marked as answer]
Okay so we just discussed this in the core meeting and we have approved it with a few points:
- we should add the supported speeds to the state attributes so frontend can read them
- We think it's best to split off the tilt speeds from this proposal, given that we first want to know how many integrations other than velux actually support setting a tilt speed

**Reply by wollew** [reply]
I updated the "simplified proposal" and removed everything related to tilt and added a short section on state attributes.

**Comment by karwosts** [comment]
I think some thought is needed as to how this can be used by frontend. 

There are lots of native places in the UI where you can click that will start a movement (buttons, toggle actions, sliders, preset selection). These will always just call move with no speed set.

Does user need a way to designate their preferred "default speed" per entity?

In the frontend more-info, we could show a dropdown of speeds, but the selection isn't saveable anywhere. So it would always be reset to "default" every time you open the more-info. We could have if user set the dropdown, and then drag the cover handle, then we call move with the selected speed. But it might be annoying that you have to reselect this every single time you open the more-info, since we will forget what you selected every time you close the dialog. And no frontend cards will respect the selection that was set there.

**Reply by wollew** [reply]
IMHO it would make sense to have a persisted default speed per-entity for the reasons you listed. But my understanding of the architecture meeting's outcome was that it is not wanted. I don't know the reasoning behind it though. Of course one can start with this minimal change now and then see what is actually needed. But thinking about the frontend for this pretty soon the conclusion will be, that a decent UX can probably be only achieved with some kind of persistence.

**Reply by karwosts** [reply]
Yeah I would agree I think the minimal change is currently fine for now, was just thinking forward to the next iteration.

**Reply by gjohansson-ST** [reply]
I think we can use entity options as a next step so the user can choose what default they want. 
Would require a discussion but should be pretty straightforward and leaves the user in charge what they prefer.

**Reply by Gunni** [reply]
> I think we can use entity options as a next step so the user can choose what default they want. 
> Would require a discussion but should be pretty straightforward and leaves the user in charge what they prefer. 

Don't make it so that my automations need to set a speed, then change the position, wait the whole time blocking the automation from doing other things, just to then set the speed back to the default.

I just want to say go to position x using speed y.

**Reply by gjohansson-ST** [reply]
That's not the case, with entity options we can provide a default (that is decided by the user) so when a speed is not provided, it uses that.
